data/model-db/generate_historical_data.R

```diff
@@ -0,0 +1,778 @@
+  # select for chinook 
+  filter(!is.na(date), species %in% c("chinook", "not recorded", "unknown")) |> 
+  select(date, latitude, longitude, reach, redd_id, age, age_index,
+         velocity, run, stream) |> 
+  left_join(survey_location, by = c("stream", "reach")) |>
+  select(-c(stream, reach, description)) |>
+  rename(survey_location_id = id) |>
+  # run_id
+  left_join(run, by = c("run" = "definition")) |> 
+  rename(run_id = id) |> 
+  select(-run, -description) 
+
+redd_reach <- daily_redd_raw |> 
+  group_by(stream) |> 
+  distinct(reach)
+
+try(if(any((unique(daily_redd$survey_location_id) %in% survey_location$id) == F)) 
+  stop("Missing Survey Location ID! Please fix!"))
+try(if(any((unique(daily_redd$run_id) %in% run$id) == F))
+  stop("Missing Run ID! Please fix!"))
+try(if(any(is.na(daily_redd$date)))
+  stop("Missing Date! Please fix!"))
+
+gcs_upload(daily_redd,
+           object_function = f,
+           type = "csv",
+           name = "model-db/daily_redd.csv",
+           predefinedAcl = "bucketLevel")
+
+# annual_redd -------------------------------------------------------------
+gcs_get_object(object_name = "standard-format-data/standard_annual_redd.csv",
+               bucket = gcs_get_global_bucket(),
+               saveToDisk = "data/standard-format-data/standard_annual_redd.csv",
+               overwrite = TRUE)
+annual_redd_raw <- read_csv("data/standard-format-data/standard_annual_redd.csv")
+
+annual_redd <- annual_redd_raw |> 
+  # filter out NA dates
+  filter(!is.na(year)) |> 
+  # filter to chinook, might want to do more work to look into error associated 
+  # with species misidentification
+  filter(species %in% c("chinook", "not recorded", "unknown")) |> 
+  select(-c(species)) |> 
+  rename(count = annual_redd_count) |> 
+  left_join(survey_location, by = c("stream", "reach")) |>
+  select(-c(stream, reach, description)) |>
+  rename(survey_location_id = id) |>
+  # run_id
+  left_join(run, by = c("run" = "definition")) |> 
+  rename(run_id = id) |> 
+  select(-run, -description) 
+
+try(if(any((unique(annual_redd$survey_location_id) %in% survey_location$id) == F)) 
+  stop("Missing Survey Location ID! Please fix!"))
+try(if(any((unique(annual_redd$run_id) %in% run$id) == F))
+  stop("Missing Run ID! Please fix!"))
+try(if(any(is.na(annual_redd$year)))
+  stop("Missing Year! Please fix!"))
+
+gcs_upload(annual_redd,
+           object_function = f,
+           type = "csv",
+           name = "model-db/annual_redd.csv",
+           predefinedAcl = "bucketLevel")
+
+# passage_counts ----------------------------------------------------------
+gcs_get_object(object_name = "jpe-model-data/upstream_passage.csv",
+               bucket = gcs_get_global_bucket(),
+               saveToDisk = "data/model-data/upstream_passage.csv",
+               overwrite = TRUE)
+passage_raw <- read_csv("data/model-data/upstream_passage.csv")
+
+passage <- passage_raw |> 
```

Comment:
Some have 0 hours sampled but have fish...I wonder if this is an error and should be NA. These are all associated with `survey_location_id = 306`

Reply:
Thanks for flagging this. I made an issue for it but don't want it to hold up adding data to db.